chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -115,6 +115,99 @@ Array Transpose(const Array& a, const OptionalAxes& axes) {
 
 namespace {
 
+std::vector<std::vector<int8_t>> NormalizeShape (const Array& array, int8_t value){
+    std::vector<std::vector<int8_t>> new_value(array.shape().ndim());
+    for (int8_t i = 0; i < array.shape().ndim(); ++i){
+      std::vector<int8_t> tuple = {value, value};
+      new_value[i] = tuple;
+    }
+    return new_value;
+}
+
+std::vector<std::vector<int64_t>> NormalizeShape (const Array& array, int64_t value){
+    std::vector<std::vector<int64_t>> new_value(array.shape().ndim());
+    for (int8_t i = 0; i < array.shape().ndim(); ++i){
+      std::vector<int64_t> tuple = {value, value};
+      new_value[i] = tuple;
+    }
+    return new_value;
+}
+
+// Returns an array after prepending a scalar value by pad_amount along the given axis
+Array PrependConst(const Array& a, int8_t pad_amount, Scalar value, int8_t axis = -1) {
+    if (pad_amount == 0) {
+        return a;
+    }
+
+    Shape in_shape{a.shape()};
+    Shape pad_shape;
+    for (int64_t i = 0; i < in_shape.ndim(); ++i) {
+        if (i != axis) {
+            pad_shape.emplace_back(in_shape[i]);
+        } else {
+            pad_shape.emplace_back(pad_amount);
+        }
+    }
+    std::vector<Array> concat_list;
+    concat_list.emplace_back(Full(pad_shape, value, a.dtype(), a.device()));
+    concat_list.emplace_back(a);
+    return Concatenate(concat_list, axis);
+}
+
+// Returns an array after appending a scalar value by pad_amount along the given axis
+Array AppendConst(const Array& a, int8_t pad_amount, Scalar value, int8_t axis = -1) {
+    if (pad_amount == 0) {
+        return a;
+    }
+
+    Shape in_shape{a.shape()};
+    Shape pad_shape;
+    for (int64_t i = 0; i < in_shape.ndim(); ++i) {
+        if (i != axis) {
+            pad_shape.emplace_back(in_shape[i]);
+        } else {
+            pad_shape.emplace_back(pad_amount);
+        }
+    }
+    std::vector<Array> concat_list;
+    concat_list.emplace_back(a);
+    concat_list.emplace_back(Full(pad_shape, value, a.dtype(), a.device()));
+    return Concatenate(concat_list, axis);
+}
+}  // namespace
+
+Array Pad(const Array& array, int8_t pad_width, PadMode mode, int64_t constant_values) {
+    std::vector<std::vector<int8_t>> new_pad_width = NormalizeShape(array, pad_width);
+    std::vector<std::vector<int64_t>> new_constant_values = NormalizeShape(array, constant_values);
+    return Pad(array, new_pad_width, mode, new_constant_values);
+}
+
+Array Pad(const Array& array, std::vector<int8_t> pad_width, PadMode mode, std::vector<int64_t> constant_values) {
+    Array newarray;
+    newarray = array.Copy();
+    if (mode == PadMode::constant) {
```

Comment:
Please use `switch` and write `CHAINERX_NEVER_REACH();` in the `default:` clause.

Reply:
Sure, done that now.